### ISSUE TITLE
[-Wunused-but-set-variable] False negative with static global variable

### ISSUE THREAD

**Issue by alejandro-colomar** [llvm/llvm-project#148361]
```c
static int foo;

int
main(void)
{
	foo = 42;
}
```
```sh
alx@devuan:~/tmp$ clang -Weverything static.c 
alx@devuan:~/tmp$ 
```

I think this should trigger some diagnostic.  The compiler is able to see that the variable is set, but never used.

Or is it too complex to gather such information about a static global?
